pkg/sdk/renderer/argo/dedicated_renderer.go

```diff
@@ -746,7 +746,8 @@ func (r *dedicatedRenderer) getInputTypeInstanceWhichSatisfyStep(step *WorkflowS
 //    - isDefined(foo,bar,baz) && isNotDefined(foo,bar,baz)
 //    - isDefined(foo,bar,baz) || isNotDefined(foo,bar,baz)
 func (*dedicatedRenderer) evaluateWhenExpression(params *mapEvalParameters, exprString string) (interface{}, error) {
-	expr, err := govaluate.NewEvaluableExpression(exprString)
+	escapeDashes := strings.Replace(exprString, "-", "\\-", -1)
```

Comment:
Please add comment, so we will know why it's needed
```suggestion
	// Dashes need to be escaped, otherwise statement is treated as expression,
	// see: https://github.com/Knetic/govaluate/tree/v3.0.0#escaping-characters
	escapeDashes := strings.Replace(exprString, "-", "\\-", -1)
```